misk/testing/src/main/kotlin/misk/testing/JUnit.kt

```diff
@@ -0,0 +1,18 @@
+package misk.testing
+
+/** Confirms that `block` throws `T`, and returns what was thrown. */
+inline fun <reified T : Throwable> assertThrows(block: () -> Unit): T {
```

Comment:
where do we draw the line on the number of assertion helpers we add to misk? don't want it turning into something like assertj (which you could just use)

Reply:
We should be comfortable adding small APIs to make Kotlin nicer. All such APIs should be candidates for upstreaming to their respective projects.

Reply:
(I'd like to submit PRs to create a guice-kotlin module with our Guice.kt stuff, for example.)